Separate tasks for css and js

It is impossible to use minify with either only css or js files.

Workaround now is pointing one of them to some empty (but existing) non-project directory.

This should be easy.
I have created a branch for it #3 
But I could not test it 

`Caused by: java.lang.NoClassDefFoundError: com/yahoo/platform/yui/compressor/CssCompressor`
It's weird I cannot even get version 1.0 to work locally. It only works if I use the published version

I'll see what happens when it's on the gradlePluginPortal.

--- a/README.md
+++ b/README.md
@@ -14,7 +14,7 @@ A simple gradle plugin to minify CSS and JavaScript files
 ### Gradle
 ```groovy
 plugins {
-  id "org.padler.gradle.minify" version "1.0"
+  id "org.padler.gradle.minify" version "1.1"
 }
 ```
 
@@ -27,7 +27,7 @@ buildscript {
     }
   }
   dependencies {
-    classpath "gradle.plugin.org.padler.gradle.minify:gradle-minify-plugin:1.0"
+    classpath "gradle.plugin.org.padler.gradle.minify:gradle-minify-plugin:1.1"
   }
 }
 

--- a/build.gradle
+++ b/build.gradle
@@ -1,19 +1,20 @@
 plugins {
     id "com.gradle.plugin-publish" version "0.10.1"
+    id 'maven-publish'
     id 'java-gradle-plugin'
     id 'groovy'
     id 'maven'
     id 'jacoco'
-    id "org.sonarqube" version "2.7"
+    id "org.sonarqube" version "2.7.1"
 }
 
 group 'org.padler.gradle.minify'
-version '1.0'
+version '1.1'
 
 sourceCompatibility = 1.8
 
 jacoco {
-    toolVersion = "0.8.3"
+    toolVersion = "0.8.4"
 }
 
 repositories {
@@ -21,7 +22,7 @@ repositories {
 }
 
 dependencies {
-    implementation 'com.yahoo.platform.yui:yuicompressor:2.4.8'
+    implementation 'com.yahoo.platform.yui:yuicompressor:2.4.7'
 
     testImplementation 'junit:junit:4.12'
 }
@@ -31,6 +32,15 @@ gradlePlugin {
         minifyPlugin {
             id = 'org.padler.gradle.minify'
             implementationClass = 'org.padler.gradle.minify.MinifyPlugin'
+        }
+    }
+}
+
+publishing {
+    publications {
+        pluginPublication(MavenPublication) {
+            groupId project.group
+            version project.version
         }
     }
 }

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.5-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists

--- a/src/main/java/org/padler/gradle/minify/MinifyTask.java
+++ b/src/main/java/org/padler/gradle/minify/MinifyTask.java
@@ -18,8 +18,10 @@ public class MinifyTask extends DefaultTask {
         Minifier cssMinifier = new CssMinifier();
         Minifier jsMinifier = new JsMinifier();
 
-        cssMinifier.minify(extension.getCssSrcDir(), extension.getCssDstDir());
-        jsMinifier.minify(extension.getJsSrcDir(), extension.getJsDstDir());
+        if (!extension.getCssSrcDir().isEmpty() && !extension.getCssDstDir().isEmpty())
+            cssMinifier.minify(extension.getCssSrcDir(), extension.getCssDstDir());
+        if (!extension.getJsSrcDir().isEmpty() && !extension.getJsDstDir().isEmpty())
+            jsMinifier.minify(extension.getJsSrcDir(), extension.getJsDstDir());
     }
 
 }

